KeyError: 'GENAI_KEY' - We require GENAI_KEY to be present even with demo_mode

Running with demo_mode true and podman compose up we still do a check and require GENAI_KEY to be in the env


```
kai_1               | Config loaded: KaiConfig(log_level='info', demo_mode=True, trace_enabled=True, incident_store=KaiConfigIncidentStorePostgreSQL(provider='postgresql', args=KaiConfigIncidentStorePostgreSQLArgs(host='kai_db', database='kai', user='kai', password='dog8code')), models=KaiConfigModels(provider='ChatIBMGenAI', args={'model_id': 'meta-llama/llama-3-70b-instruct', 'parameters': {'max_new_tokens': 2048}}, template=None, llama_header=None))
kai_1               | Logging for KAI has been initialized and the level set to INFO
kai_1               | INFO - 2024-08-09 14:52:44,897 - [           server.py:291  -                  app()] - Tracing of actions is enabled
kai_1               | INFO - 2024-08-09 14:52:44,897 - [           server.py:296  -                  app()] - DEMO_MODE is enabled. LLM responses will be cached
kai_1               | INFO - 2024-08-09 14:52:44,901 - [             psql.py:55   -             __init__()] - Connected to the PostgreSQL server.
kai_1               | INFO - 2024-08-09 14:52:44,903 - [           server.py:299  -                  app()] - Selected incident store: postgresql
kai_1               | [2024-08-09 14:52:44 +0000] [24] [ERROR] Exception in worker process
kai_1               | Traceback (most recent call last):
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
kai_1               |     worker.init_process()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/aiohttp/worker.py", line 51, in init_process
kai_1               |     super().init_process()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/workers/base.py", line 134, in init_process
kai_1               |     self.load_wsgi()
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
kai_1               |     self.wsgi = self.app.wsgi()
kai_1               |                 ^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/base.py", line 67, in wsgi
kai_1               |     self.callable = self.load()
kai_1               |                     ^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
kai_1               |     return self.load_wsgiapp()
kai_1               |            ^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
kai_1               |     return util.import_app(self.app_uri)
kai_1               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/gunicorn/util.py", line 424, in import_app
kai_1               |     app = app(*args, **kwargs)
kai_1               |           ^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/kai/kai/server.py", line 301, in app
kai_1               |     webapp["model_provider"] = ModelProvider(config.models)
kai_1               |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/kai/kai/model_provider.py", line 50, in __init__
kai_1               |     "client": Client(credentials=Credentials.from_env()),
kai_1               |                                  ^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "/opt/app-root/lib64/python3.12/site-packages/genai/credentials.py", line 54, in from_env
kai_1               |     api_key=os.environ[api_key_name or "GENAI_KEY"],
kai_1               |             ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kai_1               |   File "<frozen os>", line 685, in __getitem__
kai_1               | KeyError: 'GENAI_KEY'
```

We want folks who are evaluating Kai to be able to run in demo_mode without needing to worry about API Keys.

--- a/kai/service/llm_interfacing/model_provider.py
+++ b/kai/service/llm_interfacing/model_provider.py
@@ -52,7 +52,7 @@ class ModelProvider:
 
             case "ChatIBMGenAI":
                 model_class = LangChainChatInterface
-                if os.getenv("DEMO_MODE") == "true":
+                if os.getenv("DEMO_MODE", "false").lower() == "true":
                     api_key = os.getenv("GENAI_KEY", "dummy_value")
                     api_endpoint = os.getenv("GENAI_API", "dummy_endpoint")
                     credentials = Credentials(

--- a/kai/service/llm_interfacing/model_provider.py
+++ b/kai/service/llm_interfacing/model_provider.py
@@ -1,3 +1,5 @@
+import os
+
 from genai import Client, Credentials
 from genai.extensions.langchain.chat_llm import LangChainChatInterface
 from genai.schema import DecodingMethod
@@ -18,7 +20,6 @@ class ModelProvider:
         defaults: dict
         model_args: dict
         model_id: str
-
         # Set the model class, model args, and model id based on the provider
         match config.provider:
             case "ChatOllama":
@@ -51,9 +52,14 @@ class ModelProvider:
 
             case "ChatIBMGenAI":
                 model_class = LangChainChatInterface
-
+                if os.getenv("DEMO_MODE") == "true":
+                    credentials = Credentials(
+                        api_key="GENAI_KEY", api_endpoint="dummy_endpoint"
+                    )
+                else:
+                    credentials = Credentials.from_env()
                 defaults = {
-                    "client": Client(credentials=Credentials.from_env()),
+                    "client": Client(credentials=credentials),
                     "model_id": "ibm-mistralai/mixtral-8x7b-instruct-v01-q",
                     "parameters": {
                         "decoding_method": DecodingMethod.SAMPLE,

--- a/kai/service/llm_interfacing/model_provider.py
+++ b/kai/service/llm_interfacing/model_provider.py
@@ -53,8 +53,10 @@ class ModelProvider:
             case "ChatIBMGenAI":
                 model_class = LangChainChatInterface
                 if os.getenv("DEMO_MODE") == "true":
+                    api_key = os.getenv("GENAI_KEY", "dummy_value")
+                    api_endpoint = os.getenv("GENAI_API", "dummy_endpoint")
                     credentials = Credentials(
-                        api_key="GENAI_KEY", api_endpoint="dummy_endpoint"
+                        api_key=api_key, api_endpoint=api_endpoint
                     )
                 else:
                     credentials = Credentials.from_env()

--- a/kai/service/llm_interfacing/model_provider.py
+++ b/kai/service/llm_interfacing/model_provider.py
@@ -54,7 +54,7 @@ class ModelProvider:
                 model_class = LangChainChatInterface
                 if os.getenv("DEMO_MODE", "false").lower() == "true":
                     api_key = os.getenv("GENAI_KEY", "dummy_value")
-                    api_endpoint = os.getenv("GENAI_API", "dummy_endpoint")
+                    api_endpoint = os.getenv("GENAI_API", "")
                     credentials = Credentials(
                         api_key=api_key, api_endpoint=api_endpoint
                     )

